emulsify.drush.inc

```diff
@@ -0,0 +1,299 @@
+  $subtheme_path = 'themes';
+  if ($path = drush_get_option('path')) {
+    $subtheme_path = drush_trim_path($path);
+  }
+
+  drush_emulsify_create($name, $machine_name, $description, $subtheme_path);
+}
+
+/**
+ * Create frontend theme.
+ */
+function drush_emulsify_create($name, $machine_name, $description, $subtheme_path) {
+  $subtheme_path = drush_normalize_path(drush_get_context('DRUSH_DRUPAL_ROOT') . '/' . $subtheme_path . '/' . $machine_name);
+
+  if (!is_dir(dirname($subtheme_path))) {
+    drush_die(dt('The directory "!directory" was not found.', array('!directory' => dirname($subtheme_path))));
+  }
+
+  // Phase 1: Copy files.
+  $files_to_copy = _emulsify_get_files_to_copy();
+  _emulsify_copy_files($files_to_copy, $subtheme_path);
+
+  // Phase 2: Alter files.
+  $alterations = _emulsify_get_alterations($name, $machine_name);
+  $files_to_alter = _emulsify_get_files_to_alter();
+  _emulsify_alter_files($subtheme_path, $files_to_alter, $alterations);
+
+  // Phase 3: Rename files.
+  $files_to_rename = _emulsify_get_files_to_rename();
+  _emulsify_rename_files($subtheme_path, $machine_name, $files_to_rename);
+
+  _emulsify_notify($name, $subtheme_path);
+}
+
+/**
+ * Gets files to copy.
+ *
+ * This function supports both directories and individual files.  Alterations
+ * happen in sequential order so you can replace something that was previously
+ * replaced.
+ */
+function _emulsify_get_alterations($name, $machine_name) {
+  return array(
+    'Emulsify' => $name,
+    'emulsify' => $machine_name,
+    'hidden: true' => '',
+  );
+}
+
+/**
+ * Returns an array of files to make string replacements.
+ */
+function _emulsify_get_files_to_alter() {
+  // Slim files and directories declaration.
+  $slim_array = array(
+    'components',
+    'emulsify.info.yml',
+    'emulsify.theme',
+  );
+  // If we would like to have a bare copy we use is slim option.
+  if (drush_get_option('slim') === TRUE) {
+    return $slim_array;
+  }
+  else {
+    return array_merge($slim_array, array(
+      'templates',
+      'emulsify.breakpoints.yml',
+      'emulsify.libraries.yml',
+      'README.md',
+    ));
+  }
+}
+
+/**
+ * Gets files to copy.
+ *
+ * This function supports both directories and individual files.
+ *
+ * The following directories/files will never be copied:
+ * node_modules/
+ * github.com/
+ * composer.json
+ * LICENSE.txt
+ * emulsify.drush.inc
+ */
+function _emulsify_get_files_to_copy() {
+  // Slim files and directories declaration.
+  $slim_array = array(
+    'components',
+    'pattern-lab',
+    '.editorconfig',
+    '.gitignore',
+    '.stylelintrc',
+    'emulsify.info.yml',
+    'emulsify.theme',
+    'gulpfile.js',
+    'package.json',
+    'yarn.lock',
+  );
+  // If we would like to have a bare copy we use is slim option.
+  if (drush_get_option('slim') === TRUE) {
+    return $slim_array;
+  }
+  else {
+    return array_merge($slim_array, array(
+      'css',
+      'dist',
```

Comment:
This directory is generated after the first run of Pattern Lab. Since it is dynamically created, I don't think that we should copy this over.